Gems/Camera/Code/Source/CameraComponentController.cpp

```diff
@@ -412,6 +415,66 @@ namespace Camera
         return m_isActiveView;
     }
 
+    namespace Util
+    {
+        AZ::Vector3 GetWorldPosition(const AZ::Vector3& origin, float depth, const AzFramework::CameraState& cameraState)
+        {
+            if (depth == 0.f)
+            {
+                return origin;
+            }
+            else
+            {
+                const AZ::Vector3 rayDirection = cameraState.m_orthographic ? cameraState.m_forward : (origin - cameraState.m_position);
+                return origin + (rayDirection.GetNormalized() * depth);
+            }
+        }
+    }
+
+    AZ::Vector3 CameraComponentController::ScreenToWorld(const AZ::Vector2& screenPosition, float depth)
+    {
+        const AzFramework::ScreenPoint point{ static_cast<int>(screenPosition.GetX()), static_cast<int>(screenPosition.GetY()) };
```

Comment:
suggestion: There are useful helpers (`AzFramework::Vector2FromScreenPoint`) for this you can use which use `std::round` instead of static casting, the above can fail with essentially a value that is 3.99999f being floored instead of rounded to 4.0f

Reply:
Honestly, I'd prefer if our ScreenToWorld function accepted a float instead of int to avoid the need for casting and to account for sub-pixel precision which can really add up at far distances from the view.